kernel/Cargo.toml

```diff
@@ -47,6 +47,10 @@ thiserror = "1"
 # only for structured logging
 tracing = { version = "0.1", features = ["log"] }
 url = "2"
+# force URL to use unicode-rs instead of ICU4X otherwise we require MSRV 1.81
+# from docs: compared to ICU4X, this makes build times faster, MSRV lower, binary size larger, and
+# run-time performance slower.
+idna_adapter = "=1.1.0"
```

Comment:
We don't want to force this here, because then everyone depending on us is required to use this version. But, we can do it like I do with the home crate below. so, remove the `url = "2"` line and add:
```toml
[dependencies.url]
version = "2"
idna_adapter = "=1.1.0"
```

below

Reply:
hm it seems this doesn't force that version (just pulling in 1.2 still)